### DISCUSSION TITLE
Update the documentation of mapping such that an issue about possible code duplication is mentioned

### DISCUSSION THREAD

**Discussion by Weyaaron** [sqlalchemy/sqlalchemy#6795] [Ideas]
Dear community,
this project is really awesome, I appreciate all the hard work :)
Ever since me and my colleagues started using it we where confused about one 
particular issue:

When mapping a class using a declarative mapping, why do we have to declare both a table and a class, when the information is essentially the same? 

We never got to the bottom of this, until today, when I found out that the solution is trivial:
One of the two options is sufficient. However, this is not mentioned in the documentation(As far as I am aware)

Therefore, I would like to suggest that the documentation should be updated such that this ambiguity is addressed. 

What are your thoughts on this matter? 
Greetings,
Aaron

**Comment by zzzeek** [comment]
> Dear community,
> this project is really awesome, I appreciate all the hard work :)
> Ever since me and my colleagues started using it we where confused about one
> particular issue:
> 
> When mapping a class using a declarative mapping, why do we have to declare both a table and a class, when the information is essentially the same?

There is no need to declare both table and class.  Normal declarative mappings generate the Table for you automatically, that's what the vast majority of documentation examples show, starting with the first example in the reference documentation at https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#orm-declarative-mapping.     The "declare both a table and a class" form is known as **hybrid table** and is shown at https://docs.sqlalchemy.org/en/14/orm/declarative_tables.html#declarative-with-imperative-table-a-k-a-hybrid-declarative, this is an optional form that is preferred by some users for cleaner separation of ORM mapping and database schema.

> 
> We never got to the bottom of this, until today, when I found out that the solution is trivial:
> One of the two options is sufficient. However, this is not mentioned in the documentation(As far as I am aware)

Assuming you are looking at our docs at https://docs.sqlalchemy.org/en/14/index.html, I'm not sure how you'd come to this conclusion.   the "hybrid table" approach is only shown in a few places.      heres the [1.x tutorial](https://docs.sqlalchemy.org/en/14/orm/tutorial.html#declare-a-mapping) the [2.0 tutorial](https://docs.sqlalchemy.org/en/14/tutorial/metadata.html#declaring-mapped-classes), the [lead example in the main declarative reference docs](https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#declarative-mapping), etc., all of which show a full declarative mapping with no `Table` keyword to be found.

**Reply by Weyaaron** [reply]
> Assuming you are looking at our docs at https://docs.sqlalchemy.org/en/14/index.html, I'm not sure how you'd come to this conclusion. 

I agree, in hindsight my issue seems far fetched. One of the problems with this issue has been that there are no posts(stack overflow, ...) about it, which is obvious, since there is a obvious solution.  However, this left me wondering how such an "important" problem is missed by the wider community. Again, in hindsight, its obvious, there was never a problem to begin with. This resulted in the nagging question "What am I missing", which was left unanswered for the better part of a year. 

I'm not asking for a big change, some sort of the following is all I'm asking for:
"When used with a str as tablename", the table is created implictly".

Greetings,
Aaron

**Reply by CaselIT** [reply]
Feel free to propose an update by opening a pr.
The documentation is in the docs folder (or inline with the code if looking at the api ones) and is built with sphinx
